packtools/utils.py

```diff
@@ -459,10 +482,9 @@ def _get_all_images_to_optimise(self):
                     yield image_filename, image
 
     def _get_all_images_to_thumbnail(self):
-        path = "//graphic[@xlink:href]"
         namespaces = {"xlink": "http://www.w3.org/1999/xlink"}
-        images = self.xml_file.xpath(path, namespaces=namespaces)
-        images_parents = {image.getparent() for image in images}
+        images = self._xml_file.xpath("//graphic[@xlink:href]", namespaces=namespaces)
+        images_parents = (image.getparent() for image in images)
```

Comment:
acho que aqui seria `tuple({image.getparent() for image in images})`

Reply:
Não, aqui é um generator mesmo. 😃

Reply:
hum... mas acho que havia `{ }` para remover os duplicados

Reply:
OK, entendi seu ponto. Vou alterar.

Reply:
Feito